src/main/java/com/elasticbox/jenkins/triggers/github/PullRequestBuildHandler.java

```diff
@@ -433,14 +434,23 @@ private boolean isWhitelisted(GHUser user, GitHub gitHub) throws IOException {
         return false;
     }
 
-    private boolean isPullRequestBuild(Actionable actionable, String pullRequestUrl) {
+    private boolean isPullRequestBuildInProject(Actionable actionable, String pullRequestUrl, String projectName) {
+        String itemProjectName = null;
+        String itemRequestUrl = null;
         ParametersAction parameters = actionable.getAction(ParametersAction.class);
 
         if (parameters == null) {
             return false;
         }
         for (ParameterValue parameter: parameters.getParameters()) {
-            if ("PR_URL".equals(parameter.getName()) && pullRequestUrl.equals(parameter.getValue())) {
+            if ("PR_URL".equals(parameter.getName())) {
+                itemRequestUrl = (String) parameter.getValue();
+            } else if ("PROJECT_NAME".equals(parameter.getName())) {
+                itemProjectName = (String) parameter.getValue();
+            }
+
+            if ( (itemRequestUrl != null) && (itemProjectName != null)
```

Comment:
Unless parameters.getparameters() is a slow call or the number of parameters is usually bigger than a few thousands I prefer something less clever.

```suggestion
            itemRequestUrl = extractRequestUrl(parameters);
            itemProjectName = extractProjectName(parameters);
```

Reply:
ok